AttitudeManager/Inc/AM_DataTypes.hpp

```diff
@@ -5,17 +5,43 @@
 #ifndef ZPSW3_AM_DATATYPES_HPP
 #define ZPSW3_AM_DATATYPES_HPP
 
-typedef struct {
-    int some_data_here;
-} Position_t;
+#include "constrain.h"
+#include <stdint.h>
+#include <vector>
 
+namespace AM {
 
-enum flight_mode: int8_t {
-    fm_fatal_failure = -1,
-    fm_limp = 0,
-    fm_stabilize = 1,
-    fm_gps = 2,
-    fm_autonomous = 3
+class StateMix {
+   public:
+    const float velocity_x, velocity_y, velocity_z, pitch, roll, yaw;
+    StateMix(float velocity_x, float velocity_y, float velocity_z, float pitch,
+             float roll, float yaw)
+        : velocity_x(constrain<float, 1, -1>(velocity_x)),
+          velocity_y(constrain<float, 1, -1>(velocity_y)),
+          velocity_z(constrain<float, 1, -1>(velocity_z)),
+          pitch(constrain<float, 1, -1>(pitch)),
+          roll(constrain<float, 1, -1>(roll)),
+          yaw(constrain<float, 1, -1>(yaw)) {}
 };
 
-#endif //ZPSW3_AM_DATATYPES_HPP
+class ActuatorConfig {
+   public:
+    uint8_t channel = UINT8_MAX;
+    StateMix stateMix;
+};
+
+class ActuatorOutput {
+   public:
+    uint8_t channel;
+    float percent;
+};
+
+class AttitudeManagerInput {
```

Comment:
have u seen the discussion surrounding how data will be passed around?

Reply:
No...?